MN/Algoritmi/Metode-Iterative/SOR/SOR.m

```diff
@@ -0,0 +1,31 @@
+%  Solves linear systems A*x = B using super-over-relaxation method
+%  success - 0 = solution found / 1 = maxiter exceeded
+
+function [x success] = SOR(A, b, x0, w, tol, maxiter)
+
+    %get the number of lines and columns from matrix A
+    [n n] = size(A);
+    %initialize a column vector for the solution
+    x = zeros(n, 1);
+    %I have not found the solution yet
+    succes = 1;
+  
+    while maxiter > 0
+        maxiter--;
+    
+        for i = 1:n
+            %calculate the two sums that have to be extracted
+            sum = A(i, 1 : i - 1) * x(1 : i - 1) + A(i, i + 1 : n) * x0(i + 1 : n);
```

Comment:
poti imparti suma in old_values si new_values pentru mai multa claritate.
uita-te la PR-ul deschis numit "Metode Iterative" sau ceva de genul. Am comentat deja legat de asta.